resources/fetching/fetchers/azure/insights_batch_fetcher_test.go

```diff
@@ -0,0 +1,195 @@
+			mockInventoryError:  errors.New("mock error"),
+			cycleMetadata:       newCycle(123),
+			expected:            nil,
+			expectedErr:         true,
+		},
+
+		"empty - batch should be returned": {
+			mockSubscriptions:   []governance.Subscription{subscriptionOne},
+			mockInventoryAssets: []inventory.AzureAsset{},
+			mockInventoryError:  nil,
+			cycleMetadata:       newCycle(123),
+			expected: []fetching.ResourceInfo{
+				{
+					CycleMetadata: newCycle(123),
+					Resource: &AzureBatchResource{
+						typePair:     expectedPair,
+						Subscription: subscriptionOne,
+						Assets:       []inventory.AzureAsset{},
+					},
+				},
+			},
+			expectedErr: false,
+		},
+
+		"2 subs": {
+			mockSubscriptions: []governance.Subscription{subscriptionOne, subscriptionTwo},
+			mockInventoryAssets: []inventory.AzureAsset{
+				{Id: "id1", Name: "name1", SubscriptionId: "sub1"},
+				{Id: "id2", Name: "name2", SubscriptionId: "sub1"},
+				{Id: "id3", Name: "name3", SubscriptionId: "sub2"},
+			},
+			mockInventoryError: nil,
+			cycleMetadata:      newCycle(124),
+			expected: []fetching.ResourceInfo{
+				{
+					CycleMetadata: newCycle(124),
+					Resource: &AzureBatchResource{
+						typePair:     expectedPair,
+						Subscription: subscriptionOne,
+						Assets: []inventory.AzureAsset{
+							{Id: "id1", Name: "name1", SubscriptionId: "sub1"},
+							{Id: "id2", Name: "name2", SubscriptionId: "sub1"},
+						},
+					},
+				},
+				{
+					CycleMetadata: newCycle(124),
+					Resource: &AzureBatchResource{
+						typePair:     expectedPair,
+						Subscription: subscriptionTwo,
+						Assets: []inventory.AzureAsset{
+							{Id: "id3", Name: "name3", SubscriptionId: "sub2"},
+						},
+					},
+				},
+			},
+			expectedErr: false,
+		},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			m := azurelib.NewMockProviderAPI(t)
+
+			subscriptions := map[string]governance.Subscription{}
+			for _, s := range tc.mockSubscriptions {
+				subscriptions[s.ID] = s
+			}
+
+			m.EXPECT().GetSubscriptions(mock.Anything, mock.Anything).Return(subscriptions, nil).Once()
+
+			// check of subscriptionIDs argument
+			subIDsCheck := mock.MatchedBy(func(s []string) bool {
+				ids := lo.Keys(subscriptions)
+				sort.Strings(ids)
+				sort.Strings(s)
+				return reflect.DeepEqual(ids, s)
+			})
+			m.EXPECT().ListDiagnosticSettingsAssetTypes(mock.Anything, mock.Anything, subIDsCheck).Return(tc.mockInventoryAssets, tc.mockInventoryError).Once()
+
+			ch := make(chan fetching.ResourceInfo, 100)
+			defer close(ch)
+			f := NewAzureInsightsBatchAssetFetcher(log, ch, m)
+
+			err := f.Fetch(context.Background(), tc.cycleMetadata)
+			if tc.expectedErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
+
+			got := testhelper.CollectResources(ch)
+
+			// sort to ensure assertion
+			sortResourceInfoSlice(tc.expected)
+			sortResourceInfoSlice(got)
+
+			assert.Equal(t, tc.expected, got)
```

Comment:
Can you also please check the `GetMetadata()` and `GetElasticCommonData()` calls?

Reply:
Sure, done.